server/consumer.go

```diff
@@ -438,6 +516,9 @@ const (
 	JsFlowControlMaxPending = 32 * 1024 * 1024
 	// JsDefaultMaxAckPending is set for consumers with explicit ack that do not set the max ack pending.
 	JsDefaultMaxAckPending = 1000
+	// JsDefaultPinnedTTL is the default grace period for the pinned consumer to send a new request before a new pin
+	// is picked by a server.
+	JsDefaultPinnedTTL = 120 * time.Second
```

Comment:
This seems high?

Also 2 * time.Minute

Reply:
That is a value from ADR - we wanted to keep the default value very safe.
cc @ripienaar

Reply:
Not sure this will give a good out of the box / demo experience for failover.

Reply:
We suspect we'll be able to refine this as we use it more and as @jnmoyne starts using it.  It's just a safe default, can set to lower for demos etc

Reply:
Already using it, figured that it must be larger than your ack wait and larger than your `jetstream.PullExpiry` (fetch request's timeout).

In my library I set it to the consumer's ack wait value and make sure ack wait is no less than 6 seconds and my PullExpiry is 3 seconds, which result in a failover experience of just 6 seconds in all my failure scenarios.

Reply:
AckWait is only applicable if there is only one request pending and batch=1 I believe.

Reply:
The most common/main use for the consumer groups functionality is when you want strict in-order consumption of the messages (per subject) which means you are using max acks pending 1 so there is only one request pending (regardless of batch size). I would expect the vast majority of users of this functionality to be setting max acks pending to 1.

Reply:
I can imagine many non partitioned consumption use cases for pinning a consumer to a specific client regardless of how many ack pending it has. 

This is a feature I can see lots of use for outside of that work.

Reply:
Absolutely many will use pinning outside of partitioned consumer groups my comment was meant in that context of partitioning use case where I would expect most to use max acks pending 1. 

This relates to the initial question because (regardless of the value of max acks pending) if max acks pending is reached and the pinned client sends a next request with a timeout larger than the TTL and the TTL is longer than max wait then that pinned consumer can get unpinned (thereby flapping the pinned consumer) if ever the message takes longer than the pinned TTL to get acknowledged. Max acks pending of 1 just makes it much 'easier' for this situation to happen and to introduce flapping (if you set your TTL to less than ack wait) which is why I'm mentioning the context of most partitioning consumer groups use case and most of the time people wanting to use max acks pending 1 in those use cases.

Reply:
I guess I should conclude the thread: the default of 120s is very safe as it's much more than the default ack wait and pull expiry timeout value of 30s so no risk of introducing flapping. But I think it may be good to add a config check and throw an error if trying to create a consumer with a pinnedTTL value that is less than the ack wait value as that opens the door for introducing flapping (only if the pull expiry is also larger than the TTL, but you can only check the ack wait value at consumer creation time).